sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -99,3 +97,67 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double) extends Aggregator[IN, Double, Double] {
+  override def zero: Double = Double.PositiveInfinity
```

Comment:
Currently, I think the simplest fix is to use `java.lang.Double` as buffer and output type, so we can use `null` as zero value and make `TypedMinDouble` align with `aggregate.Min` for this corner case.

I am not sure if there is any negative effect by doing this. cc @cloud-fan for comments.